frontend/front/src/pages/Admin/home/HomeTable.js

```diff
@@ -99,32 +122,61 @@ const HomeTable = () => {
   ];
 
   const {
-    data: homesData,
-    isError: isHomesError,
+    data: fetchedHomesData,
+    isError: isFetchedHomesError,
     isLoading: isHomesDataLoading,
   } = useGetHomesQuery();
+  const {
+    data: surveyVisitsData,
+    isError: isSurveyVisitsError,
+    isLoading: isSurveyVisitsDataLoading,
+  } = useGetSurveyVisitsQuery();
+
+  let homesData = [];
 
-  if (isHomesDataLoading) {
+  if (surveyVisitsData && fetchedHomesData) {
```

Comment:
100% you want to wrap this in a `useMemo`, because otherwise it's going to recompute all of this on each rerender, regardless of whether the data has changed.

Reply:
With the code changes, this is back to 
if (isHomesDataLoading) {
    return <Loader />;
}
I am thinking this is fine as is?

Reply:
Yeah you can just leave it as is